File: 01-ordenacao/java/OrdenacaoMVC/src/model/algoritmos/SortNativo.java
package model.algoritmos;

import model.ResultadoOrdenacao;
import java.util.Collections;
import java.util.List;

/**
 * Sort nativo (Collections.sort)
 * Usa TimSort internamente. Métricas de comparações/trocas ficam zeradas
 * pois a API não as expõe
 */
public class SortNativo implements IAlgoritmoOrdenacao {

    @Override
    public String getNome() {
        return "Sort nativo (Collections.sort)";
    }

    @Override
    public ResultadoOrdenacao ordenar(List<Integer> lista) {
        long inicio = System.nanoTime();

        Collections.sort(lista);

        long tempoMs = (System.nanoTime() - inicio) / 1_000_000;

        return new ResultadoOrdenacao(0, 0, 0, tempoMs, lista.size());
    }
}

File: 01-ordenacao/csharp/OrdenacaoMVC/src/Model/Algoritmos/SortNativo.cs
using System.Diagnostics;

namespace OrdenacaoMVC.Model.Algoritmos;

/// <summary>
/// Encapsula a chamada ao <see cref="List{T}.Sort()"/> do .NET, que internamente
/// usa um algoritmo introspectivo (variação de QuickSort + HeapSort)
/// Comparações e trocas internas não são expostas pela API, portanto as
/// métricas <see cref="ResultadoOrdenacao.Comparacoes"/> e
/// <see cref="ResultadoOrdenacao.Trocas"/> ficam zeradas
///</summary>
public class SortNativo : IAlgoritmoOrdenacao
{
    public string Nome => "Sort nativo (List.Sort)";

    public ResultadoOrdenacao Ordenar(List<int> lista)
    {
        Stopwatch sw = Stopwatch.StartNew();

        lista.Sort();

        sw.Stop();

        return new ResultadoOrdenacao(
            Comparacoes: 0,
            Trocas: 0,
            Movimentacoes: 0,
            TempoMs: sw.ElapsedMilliseconds,
            TamanhoLista: lista.Count);
    }
}
